apps/meteor/client/views/room/RoomAnnouncement/RoomAnnouncement.tsx

```diff
@@ -1,23 +1,38 @@
 import { Box } from '@rocket.chat/fuselage';
-import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
+import { AnnouncementBanner } from '@rocket.chat/ui-client';
 import { useSetModal } from '@rocket.chat/ui-contexts';
-import type { MouseEvent } from 'react';
+import type { KeyboardEvent, MouseEvent } from 'react';
+import { useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
 
-import AnnouncementComponent from './AnnouncementComponent';
 import GenericModal from '../../../components/GenericModal';
 import MarkdownText from '../../../components/MarkdownText';
 
 type RoomAnnouncementParams = {
 	announcement: string;
-	announcementDetails?: () => void;
 };
 
-const RoomAnnouncement = ({ announcement, announcementDetails }: RoomAnnouncementParams) => {
+const RoomAnnouncement = ({ announcement }: RoomAnnouncementParams) => {
 	const { t } = useTranslation();
 	const setModal = useSetModal();
-	const closeModal = useEffectEvent(() => setModal(null));
-	const handleClick = (e: MouseEvent<HTMLAnchorElement>): void => {
+
+	const handleOpenAnnouncement = useCallback(() => {
```

Comment:
maybe this could be a `useEffectEvent`? Since it should only be triggered by a user interaction - click, keydown